pkg/helm/actions/upgrade_release.go

```diff
@@ -23,6 +23,12 @@ func UpgradeRelease(ns, name, url string, vals map[string]interface{}, conf *act
 		}
 		return nil, err
 	}
+
+	// Before proceeding, check if chart URL is present as an annotation
+	if chart_url, ok := rel.Chart.Metadata.Annotations["chart_url"]; url == "" && ok {
```

Comment:
Do we need to check that Annotations is not nil. Since this could be a release that was there prior the fix or it comes from helm cli? Notice that by default Annotation is nil.

Reply:
Agreed. Changing to check nil condition for Annotations.

Reply:
Do we have now a chance to have a situation when `url` is still empty?

Reply:
Yes it seems that url will be still empty, and this workaround is to pick up url from annotation.

Reply:
Updated.